Setup local test env

LongLiveCHIEF opened the pull request:
Creates a very basic ldap server in a vagrant environment.

<!---
@huboard:{"order":5.0,"milestone_order":5,"custom_state":""}
-->

LongLiveCHIEF committed:
add vagrant dev env with docker ldap container

LongLiveCHIEF committed:
add developer setup instructions in contributing.md

LongLiveCHIEF committed:
fix brew step for vagrant

LongLiveCHIEF committed:
corrected refernce vars for boxes

LongLiveCHIEF committed:
setup docker host and ldap container

LongLiveCHIEF committed:
setup docker to connect from host

LongLiveCHIEF commented:
@geekytime can you test this out for me?  Make sure you're running latest vagrant too.

geekytime commented:
:boom:

```
Chriss-MBP:impact chris$ vagrant up
Bringing machine 'ldap-server' up with 'docker' provider...
==> ldap-server: Docker host is required. One will be created if necessary...
    ldap-server: Vagrant will now create or start a local VM to act as the Docker
    ldap-server: host. You'll see the output of the `vagrant up` for this VM below.
    ldap-server:
    docker-host: Box 'LongLiveCHIEF/minimal-centos-7' could not be found. Attempting to find and install...
    docker-host: Box Provider: virtualbox
    docker-host: Box Version: >= 0
    docker-host: Loading metadata for box 'LongLiveCHIEF/minimal-centos-7'
    docker-host: URL: https://atlas.hashicorp.com/LongLiveCHIEF/minimal-centos-7
    docker-host: Adding box 'LongLiveCHIEF/minimal-centos-7' (v1.0.22) for provider: virtualbox
    docker-host: Downloading: https://atlas.hashicorp.com/LongLiveCHIEF/boxes/minimal-centos-7/versions/1.0.22/providers/virtualbox.box
    docker-host: Progress: 10% (Rate: 6657k/s, Estimated time remaining: 0:02:13    docker-host: Progress: 11% (Rate: 6758k/s, Estimated time remaining: 0:02:06    docker-host: Progress: 12% (Rate: 6672k/s, Estimated time remaining: 0:02:00    docker-host: Progress: 13% (Rate: 6694k/s, Estimated time remaining: 0:01:55    docker-host: Progress: 14% (Rate: 6656k/s, Estimated time remaining: 0:01:52    docker-host: Progress: 15% (Rate: 6652k/s, Estimated time remaining: 0:01:49    docker-host: Progress: 16% (Rate: 6630k/s, Estimated time remaining: 0:01:46    docker-host: Progress: 17% (Rate: 6700k/s, Estimated time remaining: 0:01:42    docker-host: Progress: 18% (Rate: 6684k/s, Estimated time remaining: 0:01:39    docker-host: Progress: 19% (Rate: 6729k/s, Estimated time remaining: 0:01:37    docker-host: Progress: 20% (Rate: 6748k/s, Estimated time remaining: 0:01:35    docker-host: Progress: 21% (Rate: 6787k/s, Estimated time remaining: 0:01:32    docker-host: Progress: 23% (Rate: 6739k/s, Estimated time remaining: 0:01:30    docker-host: Progress: 24% (Rate: 6700k/s, Estimated time remaining: 0:01:28    docker-host: Progress: 25% (Rate: 6662k/s, Estimated time remaining: 0:01:26    docker-host: Progress: 26% (Rate: 6643k/s, Estimated time remaining: 0:01:25    docker-host: Progress: 27% (Rate: 6576k/s, Estimated time remaining: 0:01:23    docker-host: Progress: 28% (Rate: 6553k/s, Estimated time remaining: 0:01:21    docker-host: Progress: 29% (Rate: 6627k/s, Estimated time remaining: 0:01:19    docker-host: Progress: 30% (Rate: 6633k/s, Estimated time remaining: 0:01:17    docker-host: Progress: 31% (Rate: 6710k/s, Estimated time remaining: 0:01:16    docker-host: Progress: 32% (Rate: 6752k/s, Estimated time remaining: 0:01:14    docker-host: Progress: 33% (Rate: 6739k/s, Estimated time remaining: 0:01:13    docker-host: Progress: 34% (Rate: 6641k/s, Estimated time remaining: 0:01:11    docker-host: Progress: 35% (Rate: 6665k/s, Estimated time remaining: 0:01:10    docker-host: Progress: 36% (Rate: 6599k/s, Estimated time remaining: 0:01:09    docker-host: Progress: 37% (Rate: 6611k/s, Estimated time remaining: 0:01:07    docker-host: Progress: 38% (Rate: 6624k/s, Estimated time remaining: 0:01:05    docker-host: Progress: 40% (Rate: 6702k/s, Estimated time remaining: 0:01:04    docker-host: Progress: 41% (Rate: 6672k/s, Estimated time remaining: 0:01:03    docker-host: Progress: 42% (Rate: 6668k/s, Estimated time remaining: 0:01:02    docker-host: Progress: 43% (Rate: 6601k/s, Estimated time remaining: 0:01:00    docker-host: Progress: 44% (Rate: 6652k/s, Estimated time remaining: 0:00:59    docker-host: Progress: 45% (Rate: 6616k/s, Estimated time remaining: 0:00:57    docker-host: Progress: 46% (Rate: 6780k/s, Estimated time remaining: 0:00:56    docker-host: Progress: 47% (Rate: 6982k/s, Estimated time remaining: 0:00:55    docker-host: Progress: 48% (Rate: 7052k/s, Estimated time remaining: 0:00:53    docker-host: Progress: 49% (Rate: 6982k/s, Estimated time remaining: 0:00:52    docker-host: Progress: 50% (Rate: 6966k/s, Estimated time remaining: 0:00:51    docker-host: Progress: 51% (Rate: 6841k/s, Estimated time remaining: 0:00:50    docker-host: Progress: 52% (Rate: 6739k/s, Estimated time remaining: 0:00:48    docker-host: Progress: 53% (Rate: 6502k/s, Estimated time remaining: 0:00:47    docker-host: Progress: 55% (Rate: 6531k/s, Estimated time remaining: 0:00:46    docker-host: Progress: 56% (Rate: 6531k/s, Estimated time remaining: 0:00:45    docker-host: Progress: 56% (Rate: 6281k/s, Estimated time remaining: 0:00:44    docker-host: Progress: 58% (Rate: 6390k/s, Estimated time remaining: 0:00:43    docker-host: Progress: 59% (Rate: 6646k/s, Estimated time remaining: 0:00:42    docker-host: Progress: 60% (Rate: 6652k/s, Estimated time remaining: 0:00:40    docker-host: Progress: 61% (Rate: 6710k/s, Estimated time remaining: 0:00:39    docker-host: Progress: 62% (Rate: 7040k/s, Estimated time remaining: 0:00:38    docker-host: Progress: 63% (Rate: 6934k/s, Estimated time remaining: 0:00:37    docker-host: Progress: 64% (Rate: 6737k/s, Estimated time remaining: 0:00:36    docker-host: Progress: 65% (Rate: 6766k/s, Estimated time remaining: 0:00:35    docker-host: Progress: 66% (Rate: 6723k/s, Estimated time remaining: 0:00:33    docker-host: Progress: 67% (Rate: 6611k/s, Estimated time remaining: 0:00:32    docker-host: Progress: 68% (Rate: 6524k/s, Estimated time remaining: 0:00:31    docker-host: Progress: 69% (Rate: 6686k/s, Estimated time remaining: 0:00:30    docker-host: Progress: 71% (Rate: 6636k/s, Estimated time remaining: 0:00:29    docker-host: Progress: 72% (Rate: 6684k/s, Estimated time remaining: 0:00:28    docker-host: Progress: 73% (Rate: 6732k/s, Estimated time remaining: 0:00:27    docker-host: Progress: 74% (Rate: 6806k/s, Estimated time remaining: 0:00:26    docker-host: Progress: 75% (Rate: 6678k/s, Estimated time remaining: 0:00:25    docker-host: Progress: 76% (Rate: 6659k/s, Estimated time remaining: 0:00:24    docker-host: Progress: 77% (Rate: 6572k/s, Estimated time remaining: 0:00:22    docker-host: Progress: 78% (Rate: 6150k/s, Estimated time remaining: 0:00:22    docker-host: Progress: 79% (Rate: 6211k/s, Estimated time remaining: 0:00:21    docker-host: Progress: 80% (Rate: 6371k/s, Estimated time remaining: 0:00:19    docker-host: Progress: 81% (Rate: 6400k/s, Estimated time remaining: 0:00:18    docker-host: Progress: 82% (Rate: 6451k/s, Estimated time remaining: 0:00:17    docker-host: Progress: 83% (Rate: 6636k/s, Estimated time remaining: 0:00:16    docker-host: Progress: 84% (Rate: 6678k/s, Estimated time remaining: 0:00:15    docker-host: Progress: 85% (Rate: 6640k/s, Estimated time remaining: 0:00:14    docker-host: Progress: 86% (Rate: 6668k/s, Estimated time remaining: 0:00:13    docker-host: Progress: 87% (Rate: 6662k/s, Estimated time remaining: 0:00:12    docker-host: Progress: 88% (Rate: 6838k/s, Estimated time remaining: 0:00:11    docker-host: Progress: 89% (Rate: 6812k/s, Estimated time remaining: 0:00:10    docker-host: Progress: 91% (Rate: 6793k/s, Estimated time remaining: 0:00:09    docker-host: Progress: 92% (Rate: 6704k/s, Estimated time remaining: 0:00:08    docker-host: Progress: 93% (Rate: 6563k/s, Estimated time remaining: 0:00:07    docker-host: Progress: 94% (Rate: 6582k/s, Estimated time remaining: 0:00:05    docker-host: Progress: 95% (Rate: 6630k/s, Estimated time remaining: 0:00:04    docker-host: Progress: 96% (Rate: 6592k/s, Estimated time remaining: 0:00:03    docker-host: Progress: 97% (Rate: 6480k/s, Estimated time remaining: 0:00:02    docker-host: Progress: 98% (Rate: 6393k/s, Estimated time remaining: 0:00:01    docker-host: Progress: 99% (Rate: 6438k/s, Estimated time remaining: --:--:-    docker-host: Successfully added box 'LongLiveCHIEF/minimal-centos-7' (v1.0.22) for 'virtualbox'!
    docker-host: Preparing master VM for linked clones...
    docker-host: This is a one time operation. Once the master VM is prepared,
    docker-host: it will be used as a base for linked clones, making the creation
    docker-host: of new VMs take milliseconds on a modern system.
    docker-host: Importing base box 'LongLiveCHIEF/minimal-centos-7'...
    docker-host: Cloning VM...
    docker-host: Matching MAC address for NAT networking...
    docker-host: Checking if box 'LongLiveCHIEF/minimal-centos-7' is up to date...
    docker-host: Setting the name of the VM: docker_docker-host_1457281833331_49714
    docker-host: Clearing any previously set network interfaces...
    docker-host: Preparing network interfaces based on configuration...
    docker-host: Adapter 1: nat
    docker-host: Adapter 2: hostonly
    docker-host: Forwarding ports...
    docker-host: 2375 (guest) => 2377 (host) (adapter 1)
    docker-host: 22 (guest) => 2222 (host) (adapter 1)
    docker-host: Booting VM...
    docker-host: Waiting for machine to boot. This may take a few minutes...
    docker-host: SSH address: 127.0.0.1:2222
    docker-host: SSH username: vagrant
    docker-host: SSH auth method: private key
    docker-host: Warning: Remote connection disconnect. Retrying...
    docker-host:
    docker-host: Vagrant insecure key detected. Vagrant will automatically replace
    docker-host: this with a newly generated keypair for better security.
    docker-host:
    docker-host: Inserting generated public key within guest...
    docker-host: Removing insecure key from the guest if it's present...
    docker-host: Key inserted! Disconnecting and reconnecting using new SSH key...
    docker-host: Machine booted and ready!
    docker-host: Checking for guest additions in VM...
    docker-host: Configuring and enabling network interfaces...
    docker-host: Mounting shared folders...
    docker-host: /vagrant => /Users/chris/dev/impact/docker
    docker-host: Running provisioner: file...
/opt/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/communicators/ssh/communicator.rb:271:in `initialize': No such file or directory @ rb_sysopen - /Users/chris/dev/impact/vagrant/docker.repo (Errno::ENOENT)
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/communicators/ssh/communicator.rb:271:in `open'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/communicators/ssh/communicator.rb:271:in `block in upload'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/communicators/ssh/communicator.rb:630:in `block in scp_connect'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/communicators/ssh/communicator.rb:312:in `connect'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/communicators/ssh/communicator.rb:628:in `scp_connect'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/communicators/ssh/communicator.rb:265:in `upload'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/provisioners/file/provisioner.rb:13:in `block in provision'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/provisioners/file/provisioner.rb:5:in `tap'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/provisioners/file/provisioner.rb:5:in `provision'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/provision.rb:133:in `run_provisioner'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:95:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builder.rb:116:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `block in run'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/util/busy.rb:19:in `busy'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `run'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/environment.rb:473:in `hook'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/provision.rb:121:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/provision.rb:121:in `block in call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/provision.rb:103:in `each'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/provision.rb:103:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/providers/virtualbox/action/clear_forwarded_ports.rb:15:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/providers/virtualbox/action/set_name.rb:50:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/providers/virtualbox/action/clean_machine_folder.rb:17:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/providers/virtualbox/action/check_accessible.rb:18:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builder.rb:116:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `block in run'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/util/busy.rb:19:in `busy'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `run'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/call.rb:53:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builder.rb:116:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `block in run'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/util/busy.rb:19:in `busy'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `run'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/call.rb:53:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builder.rb:116:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `block in run'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/util/busy.rb:19:in `busy'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `run'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/call.rb:53:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/box_check_outdated.rb:78:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/config_validate.rb:25:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/providers/virtualbox/action/check_virtualbox.rb:17:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/providers/virtualbox/action/match_mac_address.rb:19:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/providers/virtualbox/action/discard_state.rb:15:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/providers/virtualbox/action/import.rb:44:in `clone'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/providers/virtualbox/action/import.rb:11:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/providers/virtualbox/action/prepare_clone_snapshot.rb:32:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/prepare_clone.rb:15:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/providers/virtualbox/action/import_master.rb:36:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/providers/virtualbox/action/customize.rb:40:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/providers/virtualbox/action/check_accessible.rb:18:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builder.rb:116:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `block in run'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/util/busy.rb:19:in `busy'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `run'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/call.rb:53:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/config_validate.rb:25:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/handle_box.rb:56:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builder.rb:116:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `block in run'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/util/busy.rb:19:in `busy'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `run'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/call.rb:53:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/providers/virtualbox/action/check_virtualbox.rb:17:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builder.rb:116:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `block in run'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/util/busy.rb:19:in `busy'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `run'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/machine.rb:224:in `action_raw'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/machine.rb:199:in `block in action'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/environment.rb:561:in `lock'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/machine.rb:185:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/machine.rb:185:in `action'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/providers/docker/action/host_machine.rb:59:in `block in setup_host_machine'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/machine.rb:537:in `block in with_ui'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/machine.rb:533:in `synchronize'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/machine.rb:533:in `with_ui'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/providers/docker/action/host_machine.rb:58:in `setup_host_machine'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/providers/docker/action/host_machine.rb:28:in `block in call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/providers/docker/provider.rb:116:in `block (2 levels) in host_vm_lock'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/environment.rb:561:in `lock'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/providers/docker/provider.rb:115:in `block in host_vm_lock'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/providers/docker/provider.rb:114:in `synchronize'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/providers/docker/provider.rb:114:in `host_vm_lock'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/providers/docker/action/host_machine.rb:27:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/config_validate.rb:25:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/handle_box.rb:25:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builder.rb:116:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `block in run'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/util/busy.rb:19:in `busy'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `run'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/call.rb:53:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builder.rb:116:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `block in run'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/util/busy.rb:19:in `busy'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `run'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/machine.rb:224:in `action_raw'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/machine.rb:199:in `block in action'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/environment.rb:561:in `lock'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/machine.rb:185:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/machine.rb:185:in `action'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/batch_action.rb:82:in `block (2 levels) in run'
```

geekytime commented:
That's with a fresh install of Virtualbox and Vagrant on my MacBook.

geekytime commented:
I went back and ran `brew install ruby` to make sure it wasn't trying to use the outdated Ruby that comes with OSX.

Now I get this whenever I try to run any Vagrant commands, from `vagrant up` to `vagrant destroy --force`:

```
Chriss-MBP:impact chris$ vagrant up
Bringing machine 'ldap-server' up with 'docker' provider...
==> ldap-server: Docker host is required. One will be created if necessary...
    ldap-server: Docker host VM is already ready.
==> ldap-server: Syncing folders to the host VM...
    docker-host: Mounting shared folders...
    docker-host: /var/lib/docker/docker_1457283245_46156 => /Users/chris/dev/impact
A Docker command executed by Vagrant didn't complete successfully!
The command run along with the output from the command is shown
below.

Command: "docker" "ps" "-a" "-q" "--no-trunc"

Stderr: bash: line 2: docker: command not found

Stdout:
```

I tried `rm -rf .vagrant`, but it didn't help. Still can't run any Vagrant commands.

geekytime commented:
Also, most of this probably won't work from my work computer, due to issues with proxy, docker repos, etc. Right?

LongLiveCHIEF commented:
`cd` into the `docker` folder and run `vagrant destroy -f`, then run `vagrant up` once from within the `docker` folder.

After that, go back out to the project root, and run `vagrant up`.  If that doesn't work, then I'm going to say some choice words about vagrant, and give up on vagrant entirely. I'm starting to really not like it anymore.

LongLiveCHIEF commented:
Ok, this isn't a vagrant problem, it's actually just something I did wrong.  The way I wrote the `docker-host`'s `Vagrantfile`, it tries to find the docker install script in the wrong directory if you try to run both machines from the project root.

The instructions I gave you in the last comment _will_ work for you though, and no, there aren't any problems with work stuff, I tested it on my workstation, and it even has the new stevie settings in place, which are more restrictive than the initial configs I put in place for vagrant usage behind the firewall.

LongLiveCHIEF committed:
build rethink container on dev ip

LongLiveCHIEF committed:
fix vagrant up, merge Vagrantfiles

- moved docker definitions into one Vagrantfile
- updated README.md to include development setup
- removed Contributing.md, instructions moved to README.md

LongLiveCHIEF committed:
setting docker-host as primary machine

LongLiveCHIEF commented:
@geekytime I made some changes that should fix your issues. You'll want to:
1. `cd /path/to/impact/docker && vagrant destroy -f && rm -rf .vagrant`
2. pull branch updates
3. run `vagrant up --no-parallel`

Updates made:
- Installed a rethinkdb server
- updated developer setup instructions by migrating them to `README.md`

LongLiveCHIEF committed:
intial Dockerfile

geekytime commented:
I pulled and ran `vagrant up`, and got probably about 100 warnings and errors while it ran. Many of them were the same error: "an error occurred, a summary will be shown at the end" or whatever. Many of them were other errors and warnings. I don't have time to run it all again right now, so here's the summary at the end:

```
    docker-host: Complete!
    docker-host: Created symlink from /etc/systemd/system/multi-user.target.wants/docker.service to /usr/lib/systemd/system/docker.service.
    ldap-server: Docker host VM is already ready.
==> rethink: Syncing folders to the host VM...
    docker-host: Mounting shared folders...
    docker-host: /var/lib/docker/docker_1457306923_43284 => /Users/chris/dev/impact
==> ldap-server: Syncing folders to the host VM...
==> rethink: An error occurred. The error will be shown after all tasks complete.
    docker-host: Mounting shared folders...
    docker-host: /var/lib/docker/docker_1457306923_31903 => /Users/chris/dev/impact
==> ldap-server: Warning: When using a remote Docker host, forwarded ports will NOT be
==> ldap-server: immediately available on your machine. They will still be forwarded on
==> ldap-server: the remote machine, however, so if you have a way to access the remote
==> ldap-server: machine, then you should be able to access those ports there. This is
==> ldap-server: not an error, it is only an informational message.
==> ldap-server: Creating the container...
    ldap-server:   Name: impact_ldap-server_1457306926
    ldap-server:  Image: nickstenning/slapd
    ldap-server: Volume: /var/lib/docker/docker_1457306923_31903:/vagrant
    ldap-server:
    ldap-server: Container created: 26546b669aa1aa41
==> ldap-server: Starting container...
==> ldap-server: Provisioners will not be run since container doesn't support SSH.
An error occurred while executing multiple actions in parallel.
Any errors that occurred are shown below.

An error occurred while executing the action on the 'docker-host'
machine. Please handle this error then try again:

The box you're attempting to add doesn't support the provider
you requested. Please find an alternate box or use an alternate
provider. Double-check your requested provider to verify you didn't
simply misspell it.

If you're adding a box from HashiCorp's Atlas, make sure the box is
released.

Name: LongLiveCHIEF/minimal-centos-7
Address: https://atlas.hashicorp.com/LongLiveCHIEF/minimal-centos-7
Requested provider: [:docker]

An error occurred while executing the action on the 'rethink'
machine. Please handle this error then try again:

A Docker command executed by Vagrant didn't complete successfully!
The command run along with the output from the command is shown
below.

Command: "docker" "ps" "-a" "-q" "--no-trunc"

Stderr: Cannot connect to the Docker daemon. Is the docker daemon running on this host?

Stdout:
```

I ran `vagrant up` again immediately, as Vagrant seems to error out a lot on the first start in my experience. I didn't get any errors the second time.

How are you testing that rethink and ldap are running? I don't see any ports being forwarded back up to the host image...

geekytime commented:
Well, darn. I just realized that I missed the "--no-parallel" flag. I can try it again later with that, although this already feels pretty cobbled together and fragile. :frowning:

LongLiveCHIEF commented:
I just added the `--no-parallel` with my updates. I'm not opposed to using docker-machine, but i think the service components (databases, ldap servers, etc) will be so much easier to do with docker.

Creating the Vagrantfile also gives us another testing aspect, to make sure the various pieces of our infrastructure work on different base server OS's, and that's something we can test with via Travis using atlas builds and our vagrantfile.

The way I have it built, you'll be able to use docker-machine if you prefer, making the Vagrantfile an optional choice for each developer.

geekytime commented:
So in Hangouts you seemed frustrated by Vagrant/Docker. Would something simpler work better? I've been wanting to try [Ansible](https://www.ansible.com/)...

LongLiveCHIEF commented:
This works. Ansible isn't  a useable thing here, it would rely upon Vagrant as well.  Ansible is the same thing as chef and puppet.

My frustration was misinformed... I had a bug in my own code, which I subsequently fixed and then pushed to this PR.

geekytime commented:
Okay, I can try it again, but it still feels a little shaky. So I have to run it with this `--no-parallel` flag every time?

geekytime commented:
> Ansible isn't a useable thing here, it would rely upon Vagrant as well. Ansible is the same thing as chef and puppet.

Yeah, I'm just thinking it might be simpler than trying to do it with Docker for now, since this wasn't working, and Vagrant tends to be "bi-polar" (as a friend recently put it) at best.

LongLiveCHIEF commented:
You don't have to use vagrant at all.  It's an _optional_ feature, and I've now tested this on 4 different machines, under varying firewall and proxy conditions, and it passes just fine.

Again, I'm trying to stand up some of the mock architecture quickly so I can move forward with real work.  We're spending way too much time discussing tools in these 2 PR's. 

You have a problem with 'em, then open an issue so we can have a discussion that's not intermixed with actual constructive comments on the code involved in the PR.

geekytime commented:
> so I can move forward with real work

Dude. This _is_ real work.

geekytime commented:
> open an issue so we can have a discussion that's not intermixed with actual constructive comments on the code involved in the PR.

The PRs are here for discussion. You don't open issues against PRs, and you don't merge code unless everyone's happy with it.

LongLiveCHIEF commented:
In that caase, leave a list of what your issues are with this PR, in task bullets that I can work against.  Help sum up all the comments to make sure I didn't miss anything, (because I believe I've solved all of the comments associated with this PR thus far).

geekytime commented:
> In that caase, leave a list of what your issues are with this PR, in task bullets that I can work against. Help sum up all the comments to make sure I didn't miss anything, (because I believe I've solved all of the comments associated with this PR thus far).

I've got it running, it just takes a while. I'll let you know when it finishes up.

LongLiveCHIEF commented:
Thanks.

geekytime commented:
So again, I got errors the first time, but it started okay the second time. Looking at the errors, I think I maybe forgot to run it with `--no-parallel` the first time? :flushed: 

How do I check that rethink and ldap are installed? Should we have config tests we can run that just ping the dependencies?

LongLiveCHIEF commented:
A bunch of ways you can check:

```
vagrant ssh
docker ps
// should show two procs running, one for ldap and one for rethink
```
- You can also ping `192.168.39.39`.
- You can run `vagrant status` and verify the containers are running
- you can connect your local docker client to the docker-host, since it's a machine on your local subnet, by running `docker -d -H 192.168.39.39:2377`, and then from there you can just run docker commands on the containers on the docker-host

More work I have in the near future will give you even more options, I don't have the actual ports and stuff exposed yet, so at the moment that's the biggest thing missing from actually starting to use them.  I can add that to the PR quickly though.

geekytime commented:
I get ~~two~~ three VMs. So vagrant ssh doesn't know which one to connect to.

I could `vagrant ssh` into each one separately, but it doesn't quite line up with what you said above, so I just wanted to double-check...

geekytime commented:
The VMs are: 
- `rethink`
- `ldap-server`
- `docker-host`

LongLiveCHIEF commented:
The `docker-host` is the `primary` VM, so it will ssh into that VM when you don't specify.  otherwise, you can just `vagrant ssh docker-host`.

All of the docker containers will run _on_ that VM, and I created a local subnet, so you can do any ports you choose (even port 80, 443), by hitting `192.168.39.39`.  This will be accessible remotely, even if you're not connected to the internet.

LongLiveCHIEF committed:
add port mapping for rethink

LongLiveCHIEF committed:
created 2-step vagrant up
